src/components/common/Dropdown.tsx

```diff
@@ -0,0 +1,101 @@
+import { usePost } from "@hooks";
+import { ReactElement, useState } from "react";
+import { SlOptionsVertical } from "react-icons/sl";
+import { useNavigate } from "react-router-dom";
+
+export const Dropdown = ({ id }: { id: string }): ReactElement => {
+  const [showDropDown, setShowDropDown] = useState<boolean>(false);
+  const [showDeleteModal, setDeleteModal] = useState<boolean>(false);
```

Comment:
Instead of using state here, you can use modal params like we discussed for create and edit.

Reply:
This still needs to be refactored.